### DISCUSSION TITLE
Properties to populate from variable ( AutoPath )

### DISCUSSION THREAD

**Discussion by samuelbednarcik** [mikro-orm/mikro-orm#3346] [Q&A]
So the idea is that I want to define properties to populate in some variable, not directly when calling the populate method.

I have a controller and DTO, and in this DTO there is a populate variable that is used in many populate method calls around the app:
```typescript
export class AuctionDto {
  public static readonly populate: AutoPath<Auction, keyof Auction>[] = [
    'category',
    'organization',
    'author',
    'images',
    'mainImage',
    'socialImage',
  ];
}
```

Everything works fine until we want to add nested path, for example `organization.image`. Any idea how to achieve something like this and avoid repetition? Can't figure it out for 4 hours already.

**Comment by B4nan** [comment]
I dont think you can do this, not with the `AutoPath` type. It needs to work with generics to support nesting.

**Reply by samuelbednarcik** [reply]
Do you know about some other solution/way to do something like this? Just want to avoid repetition when calling populate method, don't care how.

**Reply by B4nan** [reply]
If you dont care about autocomplete, just put `any` in the second type arg. I am not aware of any working solution you could just plug in to have nested autocomplete support for this.
